File: bindings/lua/Polycode/Perlin.lua
class "Perlin"







function Perlin:Perlin(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycore.Perlin(unpack(arg))
	end
end

function Perlin:Get2DTiledX(x, y, t)
	local retVal = Polycore.Perlin_Get2DTiledX(self.__ptr, x, y, t)
	return retVal
end

function Perlin:Get(x, y)
	local retVal = Polycore.Perlin_Get(self.__ptr, x, y)
	return retVal
end

function Perlin:Get2D(x, y)
	local retVal = Polycore.Perlin_Get2D(self.__ptr, x, y)
	return retVal
end

function Perlin:Get3D(x, y, z)
	local retVal = Polycore.Perlin_Get3D(self.__ptr, x, y, z)
	return retVal
end

function Perlin:__delete()
	if self then Polycore.delete_Perlin(self.__ptr) end
end

File: bindings/javascript/Polycode/Perlin.js
function Perlin(octaves,freq,amp,seed) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.Perlin(octaves,freq,amp,seed)
	}
}


Duktape.fin(Perlin.prototype, function (x) {
	if (x === Perlin.prototype) {
		return;
	}
	Polycode.Perlin__delete(x.__ptr)
})

Perlin.prototype.Get2DTiledX = function(x,y,t) {
	return Polycode.Perlin_Get2DTiledX(this.__ptr, x, y, t)
}

Perlin.prototype.Get = function(x,y) {
	return Polycode.Perlin_Get(this.__ptr, x, y)
}

Perlin.prototype.Get2D = function(x,y) {
	return Polycode.Perlin_Get2D(this.__ptr, x, y)
}

Perlin.prototype.Get3D = function(x,y,z) {
	return Polycode.Perlin_Get3D(this.__ptr, x, y, z)
}
